packages/blocto-sdk/src/providers/ethereum.ts

```diff
@@ -52,114 +41,127 @@ export default class EthereumProvider
+      blockchainName: '',
+      networkType: '',
+      supportNetworkList: {},
+      switchableNetwork: {},
+    };
     this.appId = appId || DEFAULT_APP_ID;
+  }
 
-    this.switchableNetwork[this.chainId] = {
-      name: this.chain,
-      display_name: this.chain,
-      network_type: this.net,
-      wallet_web_url: this.server,
-      rpc_url: this.rpc,
+  async #getBloctoProperties(): Promise<EthereumProvider['_blocto']> {
+    if (!Object.keys(this._blocto.supportNetworkList).length) {
+      await getEvmSupport()
+        .then((result) => (this._blocto.supportNetworkList = result))
+        .catch((e) => {
+          console.error(e);
+        });
+    }
+    const {
+      chain_id,
+      name,
+      network_type,
+      blocto_service_environment,
+      display_name,
+    } = this._blocto.supportNetworkList[this.networkVersion];
+    invariant(
+      chain_id,
+      `Get blocto server failed: ${this.networkVersion} might not be supported yet.`
+    );
+    this._blocto = {
+      ...this._blocto,
+      session: this.session,
+      walletServer:
+        this.injectedWalletServer ||
+        ETH_ENV_WALLET_SERVER_MAPPING[blocto_service_environment],
+      blockchainName: name,
+      networkType: network_type,
+      switchableNetwork: {
+        [chain_id]: {
+          name,
+          display_name,
+          network_type,
+          wallet_web_url: this._blocto.walletServer,
+          rpc_url: this.rpc,
+        },
+      },
     };
+    return this._blocto;
   }
 
-  #checkAndAddNetwork({
+  async #addToSwitchable({
     chainId,
     rpcUrls,
   }: {
-    chainId: number;
+    chainId: `${number}`;
     rpcUrls: string[];
-  }): void {
-    const domain = new URL(rpcUrls[0]).hostname;
+  }): Promise<void> {
+    const { supportNetworkList } = await this.#getBloctoProperties();
     const {
       chain_id,
       name,
       display_name,
       network_type,
       blocto_service_environment,
-      rpc_endpoint_domains,
-    } = this.supportNetwork[chainId];
-    if (rpc_endpoint_domains.includes(domain)) {
-      const wallet_web_url =
-        ETH_ENV_WALLET_SERVER_MAPPING[blocto_service_environment];
-      this.switchableNetwork[chain_id] = {
-        name,
-        display_name,
-        network_type,
-        wallet_web_url,
-        rpc_url: rpcUrls[0],
-      };
-    } else {
-      console.warn(`The rpc url ${rpcUrls[0]} is not supported.`);
-    }
+    } = supportNetworkList[chainId];
+    const wallet_web_url =
+      ETH_ENV_WALLET_SERVER_MAPPING[blocto_service_environment];
+    this._blocto.switchableNetwork[chain_id] = {
+      name,
+      display_name,
+      network_type,
+      wallet_web_url,
+      rpc_url: rpcUrls[0],
+    };
   }
 
   #checkNetworkMatched(): void {
     const existedSDK = (window as any).ethereum;
     if (
       existedSDK &&
       existedSDK.isBlocto &&
-      parseInt(existedSDK.chainId, 16) !== this.chainId
+      existedSDK.chainId !== this.chainId
```

Comment:
maybe can use ```parseChainId``` function ?

Reply:
I think app sdk always use 0x-hexadecimal for chainId unlike we originally did.
But anyway add it might be better.